### ISSUE TITLE
Game control buttons are not accessible

### ISSUE THREAD

**Issue by romgrk** [online-go/online-go.com#1449]
The game control buttons are using `span` elements with onclick handlers, however span elements are not typically detected as activatable elements by accessibility tools, which renders them inaccessible for people with disabilities. The solution would be to either use `button` elements, or the ARIA attribute [role=button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role).

![Screenshot from 2021-04-11 00-24-21](https://user-images.githubusercontent.com/1423607/114292403-58a7e900-9a5c-11eb-8e10-4b7665b49c6e.png)

**Comment by anoek** [comment]
I feel like these should be changed to a slider anyways, not sure why we have 7 "buttons" there.

**Comment by romgrk** [comment]
TBH the buttons are practical, I use them often. I feel like a slider would be harder to use when going 1 move forward/backward.

**Comment by anoek** [comment]
Sounds good, I feel like I've had that conversation before and got that same feedback now that you mention it, so buttons it is :) ideally made to be accessible.

**Comment by BHydden** [comment]
Ideally we'd have both, but that brings us back to the screen real-estate dillema

**Comment by saizai** [comment]
Would adding the Aria role even make any noticeable GUI change? It seems like an easy improvement to make with near zero cost.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

I believe that the set of buttons should also be enclosed in an Aria nav role.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role

**Comment by romgrk** [comment]
No noticeable changes to the GUI except that a11y tools would be able to present things correctly to disabled (eg blind, limited arm/hand mobility, etc) users. Most of the time I'd recommend just using `button`, because you don't need to remember to add `role=button` and `tabIndex=0` (required for keyboard navigation; crucial if the user cannot use a mouse), those things are done by default for native elements. In general, any custom made element (modals, dropdowns, etc) should have ARIA attributes to make them accessible.

Also note that if the buttons are labeled only by image icons, it's important to add [invisible labels](https://stackoverflow.com/a/19758620/6303229) for screen-readers only. In this case as @saizai mentions, playing go while blind is probably rare due to the skill level required, but still nice to have just in case, and it's not too long to add. Building an accessible website takes a bit more effort but not doing it is like having a building without wheelchair access; not cool for disabled people.

**Comment by saizai** [comment]
> No noticeable changes to the GUI except that a11y tools would be able to present things correctly 

That's what I meant; sorry if I wasn't clear. I was implicitly responding to the previous comments regarding screen real estate and alternative GUI — i.e. to say that adding aria labels as you suggested wouldn't affect such considerations, while simultaneously making it more accessible.

**Comment by benjaminpjones** [comment]
Linking https://github.com/online-go/online-go.com/issues/1366

Better a11y support is definitely on the roadmap.  Getting fully there is a big project, but I don't think anyone is against these changes.

**Comment by benjaminpjones** [comment]
Keeping this open because it's a straightforward ask with clear benefit. Also adding the **Good First Issue** tag, since it's basically just an attribute on the HTML.

**Comment by benjaminpjones** [comment]
Fixed by #2227
